python/ray/tune/tune.py

```diff
@@ -439,11 +446,25 @@ def run(
         from ray.tune.schedulers import create_scheduler
         scheduler = create_scheduler(scheduler)
 
-    if issubclass(type(search_alg), Searcher):
-        search_alg = SearchGenerator(search_alg)
-
     if not search_alg:
-        search_alg = BasicVariantGenerator()
+        search_alg = BasicVariantGenerator(
+            max_concurrent=max_concurrent_trials or 0)
+    else:
+        if isinstance(search_alg, ConcurrencyLimiter):
+            logger.warning(
+                "You have specified `max_concurrent_trials="
+                f"{max_concurrent_trials}`, but the `search_alg` is already "
+                "a `ConcurrencyLimiter`. `max_concurrent_trials` will "
+                "be ignored.")
```

Comment:
Can we change this so that it raises an error if the existing concurrencylimiter max_concurrent_trials mismatch, and a warning if it matches (but has been specified twice)?

Reply:
Can do!